src_c/mixer.c

```diff
@@ -349,7 +349,27 @@ _init(int freq, int size, int channels, int chunk, char *devicename, int allowed
     if (!channels) {
         channels = request_channels;
     }
-    channels = (channels >= 2) ? 2 : 1;
+#if IS_SDLv1
+    switch (channels) {
+        case 1:
+        case 2:
+            break;
+        default:
+            PyErr_SetString(PyExc_ValueError, "'channels' must be 1 (mono) or 2 (stereo)");
```

Comment:
I made it check the arguments more strictly here. Do you think anyone relied on the old behavior?

Reply:
On the other hand, ``allowedchanges=0`` makes you expect no changes to the requested options

Reply:
Specifically, unless you set `SDL_AUDIO_ALLOW_CHANNELS_CHANGE` (or any combination including it, like `SDL_AUDIO_ALLOW_ANY_CHANGE`), I think it should fail if the value isn't valid.

Reply:
My understanding is that with SDL1.2 there is no way to enforce the no changes on the audio arguments (issue #292 is related to this regarding `format` values).

[Mix_QuerySpec](https://www.libsdl.org/projects/SDL_mixer/docs/SDL_mixer_15.html#SEC15) does not guarantee the values you get will be the ones you passed into [Mix_OpenAudio](https://www.libsdl.org/projects/SDL_mixer/docs/SDL_mixer_11.html).

> Get the actual audio format in use by the opened audio device. This may or may not match the parameters you passed to Mix_OpenAudio.

Instead of `0` as the default for `allowedchanges` maybe it should be `AUDIO_ALLOW_ANY_CHANGE` (with a note stating that is the only supported value in SDL1.2). Then the SDL1.2 code simple becomes `channels = (channels >= 2) ? 2 : 1;` (as you originally had it).